### PR TITLE
Box Component Refactor

### PR DESCRIPTION
**Pull request by sjdaily4**
This rewrites the box component using contextual components instead of block slots

https://github.com/mwisner/ember-junkdrawer/issues/3

### CODE REVIEW

File: addon/templates/ui/ui-box.hbs [hunk 1]
```diff
@@ -1,34 +1,21 @@
...
+  <h5>
+    {{yield (hash header=(component 'ui/ui-box/ui-box-header'))}}
+  </h5>
+</div>
+
+<div class="card-body {{content-modifier}}">
+  {{yield (hash body=(component 'ui/ui-box/ui-box-body'))}}
+</div>
+
+<div class="card-footer">
```
Comment (mwisner) [inline addon/templates/ui/ui-box.hbs]: I didn't realize you could actually do this.. This is sweet.

However I think we want the markup of each section in the component that represents the section.

This should make it easier to pass modifier classes and whatnot, in addition to that if for example in this setup we don't "want" a list, it's still going to render the ul wrapped around the section.

Also, in the "sub components (ui-box-footer) we can actually use the attribute `tagName` and `classNames` to set the correct tag and class names... so we shouldn't have to have the actual markup in the file at all...

File: addon/components/ui/ui-box/ui-box-list.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import Component from '@ember/component';
+
+export default Component.extend({});
```
Comment (mwisner) [inline addon/components/ui/ui-box/ui-box-list.js]: So adding something like:
```js
export default Component.extend({
tagName: 'ul',
classNames: ['list-group']
});
```

Would be better

File: addon/templates/ui/ui-box.hbs (hunk 1 above)
Comment (donaldwasserman) [reply addon/templates/ui/ui-box.hbs]: Does this work? I didn't think was supposed to work.

**Comment by mwisner** [comment]
Looking a lot better! Can you add some examples to the readme as well?

**Comment by mwisner** [comment]
@sjdaily4  anything else you'd like to add in this?

**Comment by sjdaily4** [comment]
@mwisner the only other thing would be adding the dynamic action/switch component we discussed a few days ago, but we could add that in later as a new feature so we can get these changes merged in.

**Pull request merged**